src/server/scripts/Northrend/Gundrak/boss_moorabi.cpp

```diff
@@ -190,8 +220,48 @@ class achievement_less_rabi : public AchievementCriteriaScript
         }
 };
 
+// 55163 - Mojo Frenzy
+class spell_moorabi_mojo_frenzy : public SpellScriptLoader
+{
+    public:
+        spell_moorabi_mojo_frenzy() : SpellScriptLoader("spell_moorabi_mojo_frenzy") { }
+
+        class spell_moorabi_mojo_frenzy_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_moorabi_mojo_frenzy_AuraScript);
+            
+            bool Validate(SpellInfo const* /*spell*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_MOJO_FRENZY_CAST_SPEED))
+                    return false;
+                return true;
+            }
+            
+            void HandlePeriodic(AuraEffect const* /*aurEff*/)
+            {
+                PreventDefaultAction();
+
+                Unit* owner = GetUnitOwner();
+                uint32 healthPctDamaged = uint32(100.0f - owner->GetHealthPct());
```

Comment:
Can be simplified to a single sentence (also multiplying it by 4 while still a float means more precise calc)